pkg/discovery/worker/k8s_discovery_worker.go

```diff
@@ -197,7 +195,6 @@ func (worker *k8sDiscoveryWorker) executeTask(currTask *task.Task) *task.TaskRes
 			wg.Add(1)
 			go func(w monitoring.MonitoringWorker) {
 				finishCh := make(chan struct{})
-				stopCh := make(chan struct{})
 				timeoutCh := make(chan struct{}, 1)
 				defer close(finishCh)
```

Comment:
Suggest remove this line, and replace  `finishCh <- struct{}{}` on line 228 with `close(finishCh)`. It is not a best practice to close a channel on the receiving end, as explained [here](https://go101.org/article/channel-closing.html).

Reply:
Good catch Meng. Updated. PTAL.